src/roiextractors/example_datasets/__init__.py

```diff
@@ -1,14 +1 @@
-"""Toy example ImagingExtractor and SegmentationExtractor for testing.
```

Comment:
Why go to all the effort on this PR for removing all of these? Now if I ask `roiextractors.example_datasets?` and the like I'm back to `<no docstring>`

Reply:
I removed them bc a lot of times they were a more or less exact replication of the class docstrings, so I wanted to remove extra maintenance.  I also wanted to remove the Classes section of some of the module docstrings bc it makes the api docs less intuitive. 

But I can roll this back and just keep 1-liner docstrings for every module since it helps the navigability in interactive python settings.

Reply:
> I removed them bc a lot of times they were a more or less exact replication of the class docstrings

Easy solution there is to make them more informative; e.g., describing module contents, how to import and use contents, etc.

SKlearn import style (separation into private vs. public modules) would also reduce the number of publically exposed modules that deserve such top-level documentation



> so I wanted to remove extra maintenance

But taking the time to remove them all is already more maintenance than you've had to do in many months on this 😆


> I also wanted to remove the Classes section of some of the module docstrings bc it makes the api docs less intuitive.

Surely a configurable setting for `automodule` to limit rendering of headers? Can always just use `autoclass` as well since it's usually one module per class

Reply:
IMO even if you want to change them I'd do it in a follow-up PR, not here

Goal of this PR is to just get something up and running for API docs for what we currently have, right?